neuronunit/models/backends.py

```diff
@@ -423,28 +402,127 @@ def inject_square_current(self, current):
         self.h(prefix+'amplitude=%s'%amps)
         self.h(prefix+'duration=%s'%c['duration'])
         self.h(prefix+'delay=%s'%c['delay'])
-        self.local_run()
 
+        ##
+        # Duplicated call to run?
+        self.local_run()
+        ##
     def local_run(self):
         self.h('run()')
         results={}
         results['vm'] = [float(x/1000.0) for x in copy.copy(self.neuron.h.v_v_of0.to_python())]
-        results['t'] = [float(x) for x in copy.copy(self.neuron.h.v_time.to_python())]
+        results['t'] = [float(x/1000.0) for x in copy.copy(self.neuron.h.v_time.to_python())]
```

Comment:
Is this just conversion from ms to s?  Is this value independent of the size of the time step?  If not, the time step size should appear somewhere in this calculation.  @russelljjarvis

Reply:
Actually, I am not sure if this conversion is really warranted. When the code is as previous there are sciunit errors.

It's `s` -> `ms`

n_samples = self.h.tstop/self.fixedTimeStep 
n_samples = self.h.tstop/self.h.dt

Reply:
Still working on this issue.

Reply:
I think the timestep is specified in the LEMS file and can in principle be parsed out.  Don't worry about this for now.  I will create an issue for it and deal with it later.